src/map/script.c

```diff
@@ -16294,9 +16294,56 @@ BUILDIN(pcblockmove) {
 	else
 		sd = script->rid2sd(st);
 
-	if(sd)
-		sd->state.blockedmove = flag > 0;
+	if(!sd)
+		return true;
+
+	if(flag)
+		sd->block_action |= PCBLOCK_MOVE;
+	else
+		sd->block_action &= ~PCBLOCK_MOVE;
+
+	return true;
+}
+
+BUILDIN(pcblock) {
+	int type, flag;
+	TBL_PC *sd = NULL;
+
+	type = script_getnum(st,2);
+	flag = script_getnum(st,3);
+	if( script_hasdata(st,4) )
+		sd = map->id2sd(script_getnum(st,4));
+	else
+		sd = script->rid2sd(st);
+
+	if(!sd)
+		return true;
+
+	if(!type) {
+		ShowError("buildin_pcblock: Invalid type 0.\n");
+		return false;
+	}
+
+	if(flag)
+		sd->block_action |= type;
```

Comment:
Unsafe. You don't know what value will be passed here, it can be an invalid value. (But this is irrelevant after `block_action` is changed to a bit field as suggested.)